src/reformat.c

```diff
@@ -506,17 +515,28 @@ avifResult avifImageRGBToYUV(avifImage * image, const avifRGBImage * rgb)
                 }
             }
         }
-        // Set the chroma planes to 128 if any.
+        // Set the chroma planes, if any, to the half value.
         avifPixelFormatInfo info;
         avifGetPixelFormatInfo(image->yuvFormat, &info);
         const uint32_t shiftedH = (uint32_t)(((uint64_t)image->height + info.chromaShiftY) >> info.chromaShiftY);
+        const int half = 1 << (image->depth - 1);
         if (image->yuvPlanes[AVIF_CHAN_U]) {
+            uint8_t * uPlane = image->yuvPlanes[AVIF_CHAN_U];
             const uint32_t uRowBytes = image->yuvRowBytes[AVIF_CHAN_U];
-            memset(image->yuvPlanes[AVIF_CHAN_U], 128, shiftedH * uRowBytes);
+            if (state.yuv.channelBytes > 1) {
+                avifMemset16(uPlane, half, shiftedH * uRowBytes / 2);
+            } else {
+                memset(uPlane, half, shiftedH * uRowBytes);
```

Comment:
If the row bytes (stride) is greater than the width, this memset() call is only safe if we have padding after the last row. This may not be true in general. So it would be better to replace the memset() call with a for loop that writes width pixels at a time.